R/estimate_incubation.R

```diff
@@ -0,0 +1,67 @@
+#' Function to extract the empirical incubation period distribution from line list data.
+#' Can take into account uncertain dates of exposure.
+#'
+#' @param x the linelist data (data.frame or linelist object) containing at least a column containing the exposure dates and one containing the onset dates. For exposure dates, each element can be a list containing several possible exposure dates.
+#' @param dates_exposure the name of the column containing the exposure dates.
+#' @param date_onset the name of the column containing the onset dates
+#' @return a data frame containing a column with the different incubation periods and a column containing their relative frequency
+#' @author Flavio Finger, \email{flavio.finger@lshtm.ac.uk}
+#' @export
+empirical_incubation_dist  <- function(x, dates_exposure, date_of_onset) {
+  #error checking
+  if (!is.data.frame(x)) {
+    stop("x is not a data.frame")
+  }
+
+  if (ncol(x)==0L) {
+    stop("x has no columns")
+  }
+
+  dates_exposure <- rlang::enquo(dates_exposure)
+  date_of_onset <- rlang::enquo(date_of_onset)
+
+  y <- x %>%
+    dplyr::select(!!dates_exposure, !!date_of_onset) %>%
+    dplyr::mutate(weight = purrr::map(!!dates_exposure, function(foo) 1/length(foo))) %>%
+    tidyr::unnest(!!dates_exposure, .drop  = FALSE) %>%
+    dplyr::mutate(
+      incubation_period = as.numeric(!!date_of_onset - !!dates_exposure),
+      weight = as.numeric(weight)
+    ) %>%
+    dplyr::select(incubation_period, weight) %>%
```

Comment:
R will throw a note saying that `incubation_period` and `weight` are undefined here

Reply:
I don't get why? They are columns defined within mutate...

Reply:
Even though they are defined in mutate, they aren't defined in the scope of the function and even though R knows what to do in this situation, it feels uncomfortable about it. More extensive reading on the situation: https://stackoverflow.com/q/9439256/2752888